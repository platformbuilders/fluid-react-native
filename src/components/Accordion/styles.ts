import styled from 'styled-components/native';
import { moderateScale } from 'react-native-size-matters';
import DefaultIcon from '../Icon';

export const Icon = styled(DefaultIcon).attrs({
  touchable: false,
  size: moderateScale(24),
})``;
