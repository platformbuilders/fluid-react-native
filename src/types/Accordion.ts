export type AccordionType = {
  title: string;
  content: string;
};
