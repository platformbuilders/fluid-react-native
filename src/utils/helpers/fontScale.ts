import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
