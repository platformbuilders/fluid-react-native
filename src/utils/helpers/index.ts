export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
