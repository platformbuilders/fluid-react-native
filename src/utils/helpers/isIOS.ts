import { Platform } from 'react-native';

const IOS_PLATFORM = 'ios';

export const isIOS = (): boolean => Platform.OS === IOS_PLATFORM;
