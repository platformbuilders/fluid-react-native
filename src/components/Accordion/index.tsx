import React, { useState } from 'react';
import { Text, View } from 'react-native';
import Accordion from 'react-native-collapsible/Accordion';
import Markdown from 'react-native-markdown-display';
import { head } from 'lodash';
import If from '../If';
import { AccordionType } from '../../types';
import { Icon } from './styles';

type Props = {
  StyledTitle?: JSX.Element | any;
  StyledBody?: JSX.Element | any;
  StyledHeader?: JSX.Element | any;
  StyledContent?: JSX.Element | any;
  data: AccordionType[];
  hasIcon?: boolean;
  iconUpName?: string;
  iconDownName?: string;
  activeIconColor?: string;
  inactiveIconColor?: string;
  isMarkdown?: boolean;
  onChange?(title: string): void;
};

const AccordionContainer: React.FC<Props> = ({
  StyledTitle = Text,
  StyledBody = Text,
  StyledHeader = View,
  StyledContent = View,
  data,
  hasIcon = false,
  iconUpName = 'chevron-up',
  iconDownName = 'chevron-down',
  activeIconColor,
  inactiveIconColor,
  isMarkdown = false,
  onChange,
}) => {
  const [activeSections, setActiveSections] = useState([]);

  const isActive = (section: AccordionType) => {
    const element = (item: AccordionType) => item.title === section.title;
    const index = data.findIndex(element);
    return index.toString() === activeSections.join('');
  };

  const getIconName = (section: AccordionType) =>
    isActive(section) ? iconUpName : iconDownName;

  type SectionType = {
    title: string;
    content: string;
  };

  const renderHeader = (section: SectionType) => {
    return (
      <StyledHeader isActive={isActive(section)}>
        <StyledTitle isActive={isActive(section)}>{section.title}</StyledTitle>
        <If condition={hasIcon}>
          <Icon
            name={getIconName(section)}
            color={isActive(section) ? activeIconColor : inactiveIconColor}
            id={iconUpName}
            accessibility={isActive(section) ? 'Fechar' : 'Abrir'}
          />
        </If>
      </StyledHeader>
    );
  };

  const renderContent = (section: SectionType) => {
    return (
      <StyledContent>
        <If condition={isMarkdown}>
          <Markdown>{section.content}</Markdown>
        </If>
        <If condition={!isMarkdown}>
          <StyledBody>{section.content}</StyledBody>
        </If>
      </StyledContent>
    );
  };

  const handleChange = (activeSectionsArray: []) => {
    const index: number = head(activeSectionsArray) || 0;
    if (onChange) onChange(data[index].title);
    setActiveSections(activeSectionsArray);
  };

  return (
    <Accordion
      sections={data}
      activeSections={activeSections}
      renderHeader={renderHeader}
      renderContent={renderContent}
      onChange={handleChange}
    />
  );
};

export default AccordionContainer;
