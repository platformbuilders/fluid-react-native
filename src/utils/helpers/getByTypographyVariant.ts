import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): number => {
  const fontSize = getTheme(`${props.variant}.fontSize`)(props) || 16;
  return scaledFontSize(fontSize);
};

export const getLineHeight = (props: any): number => {
  const lineHeight = getTheme(`${props.variant}.lineHeight`)(props) || 22;
  return scaledFontSize(lineHeight);
};
