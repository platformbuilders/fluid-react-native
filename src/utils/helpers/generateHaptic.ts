import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
