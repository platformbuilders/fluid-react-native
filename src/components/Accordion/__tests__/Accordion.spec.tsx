import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import Accordion from '..';
import { theme } from '../../../test/helpers';

const mockData = [
  {
    title: 'Mock',
    content: 'Mock Content',
  },
];

describe('<Accordion />', () => {
  it('should render Accordion', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Accordion data={mockData} />
      </ThemeProvider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
