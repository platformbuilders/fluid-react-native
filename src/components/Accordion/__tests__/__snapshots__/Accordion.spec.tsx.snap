// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render Accordion 1`] = `
<View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        isActive={false}
      >
        <Text
          isActive={false}
        >
          Mock
        </Text>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        forwardedRef={[Function]}
        onLayout={[Function]}
        style={Object {}}
      >
        <View>
          <Text>
            Mock Content
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
